File: Kartik/Program6/AESEncryptionExample.java
import javax.crypto.Cipher;
import javax.crypto.spec.SecretKeySpec;
import java.util.Base64;
import java.util.Scanner;

public class AESEncryptionExample {

    private static final String AES_KEY = "ThisIsASecretKey"; // 128 bit key

    public static void main(String[] args) {
        System.out.println("Enter The Text to Encrypt:");
        Scanner sc = new Scanner(System.in);
        String originalText = sc.nextLine();
        System.out.println("Original Text: " + originalText);

        try {
            byte[] encryptedText = encrypt(originalText);
            System.out.println("Encrypted Text: " + Base64.getEncoder().encodeToString(encryptedText));

            String decryptedText = decrypt(encryptedText);
            System.out.println("Decrypted Text: " + decryptedText);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public static byte[] encrypt(String plainText) throws Exception {
        SecretKeySpec keySpec = new SecretKeySpec(AES_KEY.getBytes(), "AES");
        Cipher cipher = Cipher.getInstance("AES");
        cipher.init(Cipher.ENCRYPT_MODE, keySpec);
        return cipher.doFinal(plainText.getBytes());
    }

    public static String decrypt(byte[] cipherText) throws Exception {
        SecretKeySpec keySpec = new SecretKeySpec(AES_KEY.getBytes(), "AES");
        Cipher cipher = Cipher.getInstance("AES");
        cipher.init(Cipher.DECRYPT_MODE, keySpec);
        byte[] decryptedBytes = cipher.doFinal(cipherText);
        return new String(decryptedBytes);
    }
}

File: Kartik/Program6/AESEncryptionExample.c
#include <stdio.h>
#include <string.h>
#include <openssl/aes.h>

int main() {
    unsigned char key[] = "MySecretKey"; // 16-byte key
    unsigned char iv[] = "MyInitVector"; // 16-byte IV
    unsigned char plaintext[] = "Hello, world!";
    unsigned char ciphertext[sizeof(plaintext)];
    unsigned char decryptedtext[sizeof(plaintext)];

    // Create AES key and IV
    AES_KEY aesKey;
    AES_set_encrypt_key(key, 128, &aesKey);

    // Encrypt the plaintext
    AES_cbc_encrypt(plaintext, ciphertext, sizeof(plaintext), &aesKey, iv, AES_ENCRYPT);
    printf("Encrypted text: ");
    for(int i=0; i<sizeof(ciphertext); i++) {
        printf("%02x", ciphertext[i]);
    }
    printf("\n");

    // Decrypt the ciphertext
    AES_set_decrypt_key(key, 128, &aesKey);
    AES_cbc_encrypt(ciphertext, decryptedtext, sizeof(ciphertext), &aesKey, iv, AES_DECRYPT);
    printf("Decrypted text: %s\n", decryptedtext);

    return 0;
}
//gcc -o aes_example aes_example.c -lssl -lcrypto

